packages/core/src/app/customer/StripeGuestForm.tsx

```diff
@@ -0,0 +1,249 @@
+                stripeupe: {
+                    container: 'stripeupeLink',
+                    onEmailChange: setEmailCallback,
+                    isLoading: handleLoading,
+                    getStyles: getStripeStyles,
+                    gatewayId: 'stripeupe',
+                    methodId: 'card',
+                },
+        })};
+
+    useEffect(() => {
+        stripeInitialize();
+        return () => stripeDeinitialize();
+    }, []);
+
+    const renderField = useCallback((fieldProps: FieldProps<boolean>) => (
+        <SubscribeField
+            { ...fieldProps }
+            requiresMarketingConsent={ requiresMarketingConsent }
+        />
+    ), [
+        requiresMarketingConsent,
+    ]);
+
+    const buttonText = authentication && !isNewAuth? 'customer.continue_as_stripe_customer_action' : continueAsGuestButtonLabelId;
+
+    return (
+        <>
+            <div className="checkout-form">
+                <LoadingOverlay
+                    hideContentWhenLoading
+                    isLoading={ isStripeLoading }
+                >
+                    <Fieldset
+                        legend={ !authentication &&
+                            <Legend hidden>
+                                <TranslatedString id="customer.guest_customer_text"/>
+                            </Legend>
+                        }
+                    >
+                        <div className="customerEmail-container">
+                            <div className="customerEmail-body">
+                                <div id="stripeupeLink"/>
+                                <br/>
+                                { (canSubscribe || requiresMarketingConsent) && <BasicFormField
+                                    name="shouldSubscribe"
+                                    render={ renderField }
+                                /> }
+
+                                { privacyPolicyUrl && <PrivacyPolicyField
+                                    url={ privacyPolicyUrl }
+                                /> }
+                            </div>
+
+                            <div className="form-actions customerEmail-action">
+                                { (!authentication || (authentication && !isNewAuth )) && <Button
+                                    className="stripeCustomerEmail-button"
+                                    disabled={ continueAsAGuestButton }
+                                    id="stripe-checkout-customer-continue"
+                                    isLoading={ isLoading }
+                                    onClick={ handleOnClickSubmitButton }
+                                    testId="stripe-customer-continue-as-guest-button"
+                                    type="submit"
+                                    variant={ ButtonVariant.Primary }
+                                >
+                                    <TranslatedString id={ buttonText }/>
+                                </Button> }
```

Comment:
Once we are able to get `stripeUpeLinkEnabled` and `authentication` from `checkout-sdk-js`, we can add this area to the current `<GuestForm/>`. In this way, we can avoid having a custom guest form for Stripe.

Reply:
Hey @bc-peng, we followed your suggestions and got rid of the stripeGuestForm https://github.com/bigcommerce/checkout-js/pull/1017/commits/c247ba279c5aac133034f67164b350c12162f779, please let us know if you agree with the approach